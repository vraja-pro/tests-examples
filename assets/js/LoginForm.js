import React, { useState } from 'react';

const LoginForm = () => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [loggedIn, setLoggedIn] = useState(false);

  const handleFormSubmit = (event) => {
    event.preventDefault();
    // Here you could perform validation, authentication, etc.
    // For simplicity, let's just consider it logged in if both fields are filled.
    if (username.trim() !== '' && password.trim() !== '') {
      setLoggedIn(true);
    }
  };

  return (
    <div>
      {!loggedIn ? (
        <form onSubmit={handleFormSubmit}>
          <div>
            <label htmlFor="username">Username:</label>
            <input
              type="text"
              id="username"
              value={username}
              onChange={(e) => setUsername(e.target.value)}
            />
          </div>
          <div>
            <label htmlFor="password">Password:</label>
            <input
              type="password"
              id="password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
          </div>
          <button type="submit">Login</button>
        </form>
      ) : (
        <div>
          <h2>Welcome, {username}!</h2>
          <button onClick={() => setLoggedIn(false)}>Logout</button>
        </div>
      )}
    </div>
  );
};

export default LoginForm;
