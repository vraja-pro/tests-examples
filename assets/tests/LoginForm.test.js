import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LoginForm from '../js/LoginForm';


describe('LoginForm', () => {
  it('renders login form initially', () => {
    const { getByLabelText, getByText } = render(<LoginForm />);

    expect(getByLabelText('Username:')).toBeInTheDocument();
    expect(getByLabelText('Password:')).toBeInTheDocument();
    expect(getByText('Login')).toBeInTheDocument();
  });

  it('renders welcome message after successful login', () => {
    const { getByLabelText, getByText } = render(<LoginForm />);
    const usernameInput = getByLabelText('Username:');
    const passwordInput = getByLabelText('Password:');
    const loginButton = getByText('Login');

    // Simulate user input and submit form
    fireEvent.change(usernameInput, { target: { value: 'user' } });
    fireEvent.change(passwordInput, { target: { value: 'password' } });
    fireEvent.click(loginButton);

    // Assert welcome message and logout button
    expect(getByText('Welcome, user!')).toBeInTheDocument();
    expect(getByText('Logout')).toBeInTheDocument();
  });

  it('renders login form again after logout', () => {
    const { getByText, getByLabelText } = render(<LoginForm />);
    const loginButton = getByText('Login');
   
    const usernameInput = getByLabelText('Username:');
    const passwordInput = getByLabelText('Password:');

    fireEvent.change(usernameInput, { target: { value: 'user' } });
    fireEvent.change(passwordInput, { target: { value: 'password' } });

    // Click on logout button
    fireEvent.click(loginButton);

    const logoutButton = getByText('Logout');
    // Assert logout form
    expect(logoutButton).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<LoginForm />);
    expect(container).toMatchSnapshot();
  });

});
