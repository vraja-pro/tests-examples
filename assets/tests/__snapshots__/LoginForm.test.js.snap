// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches snapshot 1`] = `
<div>
  <div>
    <form>
      <div>
        <label
          for="username"
        >
          Username:
        </label>
        <input
          id="username"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="password"
        >
          Password:
        </label>
        <input
          id="password"
          type="password"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
